src/labkey/Experiment.ts

```diff
@@ -0,0 +1,592 @@
+            return new RunGroup(json);
+        }, getOnSuccess(options), options.scope),
+        failure: getCallbackWrapper(getOnFailure(options), options.scope, true)
+    });
+}
+
+/**
+ * @hidden
+ * @private
+ */
+function getSuccessCallbackWrapper<SuccessPayload>(
+    payloadProcessor: (json: any) => SuccessPayload,
+    success: any,
+    scope?: any) {
+
+    return getCallbackWrapper((json: any, response: ExtendedXMLHttpRequest) => {
+        if (success) {
+            success.call(scope || this, payloadProcessor(json), response);
+        }
+    });
+}
+
+export interface IBaseSaveBatchOptions {
+    /**
+     * The assay protocol id
+     */
+    assayId: number
+
+    /**
+     * The name of the assay.
+     */
+    assayName?: string
+
+    /**
+     * A reference to a function to call when an error occurs. This function will be passed the following parameters:
+     */
+    failure?: ExperimentFailureCallback
+
+    /**
+     * Optional protocol name to be used for non-assay backed runs. Currently only SAMPLE_DERIVATION_PROTOCOL
+     * is supported.
+     */
+    protocolName?: string
+
+    /**
+     * The assay provider name.
+     */
+    providerName?: string
+
+    /**
+     * A scoping object for the success and failure callback functions (default to this).
+     */
+    scope?: any
+}
+
+export interface ISaveBatchOptions extends IBaseSaveBatchOptions {
+    /**
+     * A modified RunGroup.
+     */
+    batch?: RunGroup
+
+    /**
+     * The function to call when loadBatches finishes successfully.
+     */
+    success?: ExperimentSuccessCallback<RunGroup>
+}
+
+export interface ISaveBatchesOptions extends IBaseSaveBatchOptions {
+    /**
+     * The modified RunGroups.
+     */
+    batches?: Array<RunGroup>
+
+    /**
+     * The function to call when loadBatches finishes successfully.
+     */
+    success?: ExperimentSuccessCallback<Array<RunGroup>>
+}
+
+export interface ILineageOptions {
+    /**
+     * Include children in the lineage response. Defaults to true.
+     */
+    children?: boolean
+
+    /**
+     * Optional LSID of a SampleSet or DataClass to filter the response. Defaults to include all.
+     */
+    cpasType?: string
+
+    /**
+     * An optional depth argument.  Defaults to include all.
+     */
+    depth?: number
+
+    /**
+     * Optional experiment type to filter response -- either "Data", "Material", or "ExperimentRun". Defaults to include all.
+     */
+    expType?: string
+
+    /**
+     * A reference to a function to call when an error occurs.
+     */
+    failure?: () => any
+
+    /**
+     * The LSID of the seed ExpData or ExpMaterial.  Either rowId or lsid is required.
+     */
+    lsid?: string
+
+    /**
+     * Include parents in the lineage response.  Defaults to true.
+     */
+    parents?: boolean
+    
+    /**
+     * The row id of the seed ExpData or ExpMaterial.  Either rowId or lsid is required.
+     */
+    rowId?: any // TODO: string or number or both?
```

Comment:
Since this will supplant Experiment.js as it currently is used I'm going to keep it in. I will update the documentation to express that it is deprecated.

Reply:
Updated with 583b50d.